File: fund-quant-terminal/backend/app/models/wallstreet.py
# =====================================================
# 华尔街见闻 API 响应模型
# =====================================================

from typing import Any

from pydantic import BaseModel, Field


class WallStreetResponse(BaseModel):
    """华尔街见闻测试接口响应数据"""

    type: str = Field(..., description="请求类型: lives|articles|search|quote|keyword")
    data: Any = Field(..., description="华尔街见闻 API 原始 JSON 响应")

File: fund-quant-terminal/frontend/src/news/api/wallstreet.ts
// =====================================================
// 华尔街见闻 API（news 模块）
// =====================================================

import { request } from "@/api/request";

export type WallStreetCNType = "lives" | "articles" | "search" | "quote" | "keyword";

export interface WallStreetCNTestParams {
  type: WallStreetCNType;
  code?: string;
  keyword?: string;
  limit?: number;
  channel?: string;
  cursor?: number;
  save_to_db?: boolean;
}

export const wallstreetcnTest = (params: WallStreetCNTestParams) =>
  request.post("wallstreetcn/test", params, { skipLoading: true });
